pkg/network/http/ebpf_gotls.go

```diff
@@ -97,80 +97,87 @@ func (p *GoTLSProgram) Start() {
 	// and this implementation should be done for each new process found.
 
 	binPath := os.Getenv("GO_TLS_TEST")
-	p.handleNewBinary(binPath)
+	if binPath != "" {
+		p.handleNewBinary(binPath)
+	}
+}
 
+func supportedArch(arch bininspect.GoArch) bool {
+	return arch == bininspect.GoArchX86_64
 }
 
 func (p *GoTLSProgram) handleNewBinary(binPath string) {
 	f, err := os.Open(binPath)
 	if err != nil {
-		log.Errorf("could not open file %q due to %w", binPath, err)
+		log.Errorf("Could not open file %q due to %w", binPath, err)
 		return
 	}
 	defer f.Close()
 	elfFile, err := elf.NewFile(f)
 	if err != nil {
-		log.Errorf("file %q could not be parsed as elf: %w", binPath, err)
+		log.Errorf("File %q could not be parsed as elf: %w", binPath, err)
 		return
 	}
+
 	result, err := bininspect.InspectNewProcessBinary(elfFile, functionsConfig, structFieldsLookupFunctions)
 	if err != nil {
-		log.Errorf("failed inspecting binary %q: %w", binPath, err)
+		log.Errorf("Failed inspecting binary %q: %w", binPath, err)
+		return
+	}
+
+	if !supportedArch(result.Arch) {
+		log.Errorf("Got unsupported arch %q for binary %q", result.Arch, binPath)
 		return
 	}
```

Comment:
I think it can happen only once on the start of goTLS program
as if the machine architecture is arch, there is not point with checking each process by itself